### ISSUE TITLE
Renewal attempted on expired cert, crash

### ISSUE THREAD

**Issue by mengesb** [terraform-providers/terraform-provider-acme#72]
I know this won't be a terrible amount of data go to on, and reproducing this takes 91+ days.

I ran my acme plan attempt to renew a certificate expired yesterday, and the provider crashed. Naturally, since I use pTFE, I don't have the exact crash.log however I do have the terminal output:

```text
Terraform v0.11.13

Initializing plugins and modules...
2019/07/05 16:11:00 [DEBUG] Using modified User-Agent: Terraform/0.11.13 PTFE/e63ebf6
acme_certificate.acme: Modifying... (ID: https://acme-v02.api.letsencrypt.org/ac...t/0301baacc85183a04279ed465afcf62cdb83)
  certificate_domain: "dev1.app-arcus.anaplan.io" => "<computed>"
  certificate_p12:    "<sensitive>" => "<sensitive>"
  certificate_pem:    "-----BEGIN CERTIFICATE-----\nREDACTED\n-----END CERTIFICATE-----\n" => "<computed>"
  certificate_url:    "https://acme-v02.api.letsencrypt.org/acme/cert/0301baacc85183a04279ed465afcf62cdb83" => "<computed>"
  issuer_pem:         "-----BEGIN CERTIFICATE-----\nnREDACTED\n-----END CERTIFICATE-----\n" => "<computed>"
  private_key_pem:    "<sensitive>" => "<sensitive>"

Error: Error applying plan:

1 error(s) occurred:

* acme_certificate.acme: 1 error(s) occurred:

* acme_certificate.acme: unexpected EOF

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.


panic: runtime error: invalid memory address or nil pointer dereference
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0xdbc068]
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: goroutine 46 [running]:
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: github.com/terraform-providers/terraform-provider-acme/vendor/github.com/go-acme/lego/certcrypto.ParsePEMPrivateKey(0x2d5c320, 0x0, 0x0, 0xc000669880, 0x2, 0x2, 0x38)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-acme/vendor/github.com/go-acme/lego/certcrypto/crypto.go:83 +0x48
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: github.com/terraform-providers/terraform-provider-acme/vendor/github.com/go-acme/lego/certificate.(*Certifier).Renew(0xc0001304b0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x2d5c320, 0x0, 0x0, ...)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-acme/vendor/github.com/go-acme/lego/certificate/certificates.go:372 +0x58b
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: github.com/terraform-providers/terraform-provider-acme/acme.resourceACMECertificateUpdate(0xc000200c40, 0x14fe320, 0xc0001624a0, 0x24, 0x2d5dc00)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-acme/acme/resource_acme_certificate.go:461 +0x405
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: github.com/terraform-providers/terraform-provider-acme/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Apply(0xc00018b200, 0xc0000c2280, 0xc0000a8f80, 0x14fe320, 0xc0001624a0, 0xc00057ec01, 0x0, 0xc0003a6bd0)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-acme/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:292 +0x264
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: github.com/terraform-providers/terraform-provider-acme/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Apply(0xc00018b280, 0xc0000c2230, 0xc0000c2280, 0xc0000a8f80, 0xc0004c9470, 0xc0003a6bd0, 0x7fe669aba6d0)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-acme/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:285 +0x18f
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: github.com/terraform-providers/terraform-provider-acme/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Apply(0xc0004ad7c0, 0xc0000a8a00, 0xc000163e20, 0x0, 0x0)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/teamcity-agent/work/5d79fe75d4460a2f/src/github.com/terraform-providers/terraform-provider-acme/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:530 +0x57
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: reflect.Value.call(0xc000590540, 0xc00016a048, 0x13, 0x18ca7b0, 0x4, 0xc00058cf18, 0x3, 0x3, 0xc0004b8d40, 0x0, ...)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/goenv/versions/1.12.5/src/reflect/value.go:447 +0x461
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: reflect.Value.Call(0xc000590540, 0xc00016a048, 0x13, 0xc000220f18, 0x3, 0x3, 0x0, 0x0, 0x0)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/goenv/versions/1.12.5/src/reflect/value.go:308 +0xa4
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: net/rpc.(*service).call(0xc0004b8600, 0xc000172190, 0xc0004664e8, 0xc000466510, 0xc000122000, 0xc000166620, 0x150cba0, 0xc0000a8a00, 0x16, 0x150cbe0, ...)
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/goenv/versions/1.12.5/src/net/rpc/server.go:384 +0x14e
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: created by net/rpc.(*Server).ServeCodec
2019-07-05T16:11:02.463Z [DEBUG] plugin.terraform-provider-acme_v1.3.4_x4: 	/opt/goenv/versions/1.12.5/src/net/rpc/server.go:481 +0x42b
2019/07/05 16:11:02 [TRACE] root: eval: *terraform.EvalWriteState
2019-07-05T16:11:02.465Z [DEBUG] plugin: plugin process exited: path=/terraform/.terraform/plugins/linux_amd64/terraform-provider-acme_v1.3.4_x4
2019/07/05 16:11:02 [TRACE] root: eval: *terraform.EvalApplyProvisioners
2019/07/05 16:11:02 [TRACE] root: eval: *terraform.EvalIf
2019/07/05 16:11:02 [TRACE] root: eval: *terraform.EvalWriteState
2019/07/05 16:11:02 [TRACE] root: eval: *terraform.EvalWriteDiff
2019/07/05 16:11:02 [TRACE] root: eval: *terraform.EvalApplyPost
2019/07/05 16:11:02 [ERROR] root: eval: *terraform.EvalApplyPost, err: 1 error(s) occurred:

* acme_certificate.acme: unexpected EOF
2019/07/05 16:11:02 [ERROR] root: eval: *terraform.EvalSequence, err: 1 error(s) occurred:

* acme_certificate.acme: unexpected EOF
2019/07/05 16:11:02 [TRACE] [walkApply] Exiting eval tree: acme_certificate.acme
2019/07/05 16:11:02 [TRACE] dag/walk: upstream errored, not walking "provider.acme (close)"
2019/07/05 16:11:02 [TRACE] dag/walk: upstream errored, not walking "output.acme_certificate"
2019/07/05 16:11:02 [TRACE] dag/walk: upstream errored, not walking "meta.count-boundary (count boundary fixup)"
2019/07/05 16:11:02 [TRACE] dag/walk: upstream errored, not walking "root"
2019/07/05 16:11:02 [TRACE] Preserving existing state lineage "374b2ccd-61d7-0c29-f41c-03ffa14f79b6"
2019/07/05 16:11:02 [TRACE] Preserving existing state lineage "374b2ccd-61d7-0c29-f41c-03ffa14f79b6"
2019/07/05 16:11:02 [TRACE] Preserving existing state lineage "374b2ccd-61d7-0c29-f41c-03ffa14f79b6"
2019/07/05 16:11:02 [DEBUG] PUT REDACTED
2019/07/05 16:11:02 [DEBUG] plugin: waiting for all plugin processes to complete...
2019-07-05T16:11:02.614Z [WARN ] plugin: error closing client during Kill: err="connection is shut down"



!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

Terraform crashed! This is always indicative of a bug within Terraform.
A crash log has been placed at "crash.log" relative to your current
working directory. It would be immensely helpful if you could please
report the crash with Terraform[1] so that we can fix this.

When reporting bugs, please include your terraform version. That
information is available on the first line of crash.log. You can also
get it by running 'terraform --version' on the command line.

[1]: https://github.com/hashicorp/terraform/issues

!!!!!!!!!!!!!!!!!!!!!!!!!!! TERRAFORM CRASH !!!!!!!!!!!!!!!!!!!!!!!!!!!!

State versions created:
REDACTED#sv-VcJZP62tMCfA9HyL (Jul 05, 2019 09:11:01 am)
REDACTED#sv-5bpKVUsefPJRS41k (Jul 05, 2019 09:11:02 am)
```

I'm sure this has something to do with reading the certificate and maybe calculating a negative time remaining on the certificate in the provider. Of course the trace shows hints of this being lower down in lego so not sure if this is a dependency problem either.

**Comment by xyloman** [comment]
I ran into a very similar issue today but the certificate is about 30 days away from being expired. First attempt failed ran the terraform plan again and it succeed.

**Comment by vancluever** [comment]
Hey @mengesb, I haven't been able to reproduce this and the renewal test was working as of last release.

I think I've traced it back [here](https://github.com/terraform-providers/terraform-provider-acme/blob/db0b8ba51c0471c7cb4e4a386b90047a7751657f/acme/acme_structure.go#L180-L181) though, as I suspect this is creating empty `[]byte`s that are being fed into `ParsePEMPrivateKey`, causing the panics. There is a nil check farther up the stack, so hopefully this gets skipped if it's not populated. I'll put in a update for this and hopefully it will help stability.

**Comment by vancluever** [comment]
Should be fixed with the release of v1.4.0.
